src/test/kotlin/org/dif/fixtures/Fixtures.kt

```diff
@@ -690,21 +719,415 @@ class JWE {
+                    .Builder(BOB_MESSAGE_WITHOUT_CIPHERTEXT)
+                    .expectDecryptByAllKeys(true)
+                    .build()
+            ),
+
+            NegativeTestVector(
+                packedMessage = BOB_MESSAGE_WITHOUT_TAG,
+                expectedThrow = MalformedMessageException::
+                class,
+                expectedMessage = "Decrypt is failed",
+                unpackParams = UnpackParams
+                    .Builder(BOB_MESSAGE_WITHOUT_TAG)
+                    .expectDecryptByAllKeys(true)
+                    .build()
+            ),
+
+            NegativeTestVector(
+                packedMessage = BOB_MESSAGE_WITHOUT_IV,
+                expectedThrow = MalformedMessageException::
+                class,
+                expectedMessage = "Decrypt is failed",
+                unpackParams = UnpackParams
+                    .Builder(BOB_MESSAGE_WITHOUT_IV)
+                    .expectDecryptByAllKeys(true)
+                    .build()
+            ),
+
             NegativeTestVector(
                 packedMessage = BOB_DAMAGED_MESSAGE,
-                expectedThrow = MalformedMessageException::class,
+                expectedThrow = MalformedMessageException::
+                class,
                 expectedMessage = "Decrypt is failed",
                 unpackParams = UnpackParams
                     .Builder(BOB_DAMAGED_MESSAGE)
                     .expectDecryptByAllKeys(true)
                     .build()
+            ),
+
+            NegativeTestVector(
+                packedMessage = BOB_MESSAGE_UNSUPPORTED_ALG_HEADER,
+                expectedThrow = UnsupportedAlgorithm::
+                class,
+                expectedMessage = "The algorithm ECDH-ES+A255KW+XC20P is unsupported",
+                unpackParams = UnpackParams
+                    .Builder(BOB_MESSAGE_UNSUPPORTED_ALG_HEADER)
+                    .expectDecryptByAllKeys(true)
+                    .build()
+            ),
+
+            NegativeTestVector(
+                packedMessage = BOB_MESSAGE_UNSUPPORTED_ENC_HEADER,
+                expectedThrow = UnsupportedAlgorithm::
+                class,
+                expectedMessage = "The algorithm ECDH-ES+A256KW+XC202 is unsupported",
+                unpackParams = UnpackParams
+                    .Builder(BOB_MESSAGE_UNSUPPORTED_ENC_HEADER)
+                    .expectDecryptByAllKeys(true)
+                    .build()
+            ),
+
+            NegativeTestVector(
+                packedMessage = BOB_ANON_MESSAGE_RECIPIENT_KEY_IS_INVALID,
+                expectedThrow = MalformedMessageException::
+                class,
+                expectedMessage = "Decrypt is failed",
+                unpackParams = UnpackParams
+                    .Builder(BOB_ANON_MESSAGE_RECIPIENT_KEY_IS_INVALID)
+                    .expectDecryptByAllKeys(true)
+                    .build()
+            ),
+
+//            NegativeTestVector(
```

Comment:
I took a thought and got it that it's a correct behavior. Because we suspect that the message will be decrypted only the keys stored in secret manager.

Reply:
resolved